src/fastfield/writer.rs

```diff
@@ -163,3 +189,243 @@ impl FastFieldsWriter {
         Ok(())
     }
 }
+
+#[inline]
+fn columnar_numerical_value(json_number: &serde_json::Number) -> Option<NumericalValue> {
+    if let Some(num_i64) = json_number.as_i64() {
+        return Some(num_i64.into());
+    }
+    if let Some(num_u64) = json_number.as_u64() {
+        return Some(num_u64.into());
+    }
+    if let Some(num_f64) = json_number.as_f64() {
+        return Some(num_f64.into());
+    }
+    // This can happen with arbitrary precision.... but we do not handle it.
+    None
+}
+
+fn record_json_obj_to_columnar_writer(
+    doc: DocId,
+    json_obj: &serde_json::Map<String, serde_json::Value>,
+    expand_dots: bool,
+    depth_limit: usize,
+    json_path_buffer: &mut String,
+    columnar_writer: &mut columnar::ColumnarWriter,
+) {
+    for (key, child) in json_obj {
+        let len_path = json_path_buffer.len();
+        if !json_path_buffer.is_empty() {
+            json_path_buffer.push_str(JSON_PATH_SEGMENT_SEP_STR);
+        }
+        json_path_buffer.push_str(key);
+        if expand_dots {
+            // This might include the separation byte, which is ok because it is not a dot.
+            let appended_segment = &mut json_path_buffer[len_path..];
+            // The unsafe below is safe as long as b'.' and JSON_PATH_SEGMENT_SEP are
+            // valid single byte ut8 strings.
+            // By utf-8 design, they cannot be part of another codepoint.
+            replace_in_place(b'.', JSON_PATH_SEGMENT_SEP, unsafe {
+                appended_segment.as_bytes_mut()
+            });
```

Comment:
Do we replace JSON_PATH_SEGMENT_SEP with `.`, so we can query it?

Why don't we use `.` instead of JSON_PATH_SEGMENT_SEP

Reply:
that's the other way around. we replace '.' by the separator